### PR TITLE
fix for build, apply semantics

### PR DESCRIPTION
**Pull request by swiftdiaries**
Resolves: https://github.com/kubeflow/kfctl/issues/25
Resolves: https://github.com/kubeflow/kubeflow/issues/4242

Current workflow for build -> apply errors out 
```
Error: (kubeflow.error): Code 400 with message: current directory not empty, please switch directories
```
This PR is a fix for that
Adding an E2E test for this
/cc @richardsliu

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4232)

### CODE REVIEW

**Comment by swiftdiaries** [comment]
Also, adding myself as a reviewer 
/cc @kunmingg  @yanniszark @kkasravi @gabrielwen

**Comment by nrchakradhar** [comment]
@swiftdiaries kfctl_go_test was split into separate tests: [build](https://github.com/kubeflow/kubeflow/blob/master/py/kubeflow/kubeflow/ci/kfctl_go_build_test.py) and [deploy](https://github.com/kubeflow/kubeflow/blob/master/py/kubeflow/kubeflow/ci/kfctl_go_deploy_test.py)
I guess the new flag build_and_deploy is really not needed and above tests may be used.
Hope its not a digress for this PR.

File: testing/kfctl/kfctl_go_test.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from kubeflow.kubeflow.ci import kfctl_go_test_utils as kfctl_util
 from kubeflow.testing import util
 
-def test_build_kfctl_go(app_path, project, use_basic_auth, use_istio, config_path):
+def test_build_kfctl_go(app_path, project, use_basic_auth, use_istio, config_path, build_and_apply):
   """Test building and deploying Kubeflow.
 
   Args:
```
Comment (gabrielwen) [inline testing/kfctl/kfctl_go_test.py]: docs for the new arg?

Comment (swiftdiaries) [reply testing/kfctl/kfctl_go_test.py]: Added

File: bootstrap/OWNERS [hunk 2]
```diff
@@ -4,3 +4,4 @@ approvers:
   - kunmingg
   - yanniszark
 reviewers:
+  - swiftdiaries
```
[the remark this answers is not in the document]
Comment (swiftdiaries) [reply bootstrap/OWNERS]: Done.

**Pull request merged**